### PR TITLE
Add Azure SDK instrumentation

### PR DESCRIPTION
**Pull request by trask**

### CODE REVIEW

File: instrumentation/azure-core/azure-core-1.14/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/azurecore/v1_14/AzureSdkInstrumentationModule.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+      //
+      // this is a list of all classes that call one of these:
+      // * ServiceLoader.load(AfterRetryPolicyProvider.class)
+      // * ServiceLoader.load(Tracer.class)
+      return named("com.azure.core.http.policy.HttpPolicyProviders")
+          .or(named("com.azure.core.util.tracing.TracerProxy"))
+          .or(named("com.azure.cosmos.CosmosAsyncClient"))
+          .or(named("com.azure.messaging.eventhubs.EventHubClientBuilder"))
+          .or(named("com.azure.messaging.eventhubs.EventProcessorClientBuilder"))
+          .or(named("com.azure.messaging.servicebus.ServiceBusClientBuilder"));
```
Comment (trask) [inline instrumentation/azure-core/azure-core-1.14/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/azurecore/v1_14/AzureSdkInstrumentationModule.java:68]: I thought we could simplify this similar to #5216 (thanks to #5185), but an issue (#5466) was just opened about #5216 causing a regression

**Review by lmolkova — COMMENTED** [review]
As per offline discussion, we should keep suppression in 1.19 version. Otherwise looks great!

**Review by anuraaga — APPROVED** [review]
Wouldn't block it but want to point out that there seems to be quite a small span between 1.14 (8/2018) and 1.19 (1/2019) that having multiple instrumentation seems a bit overkill.

**Comment by trask** [comment]
> Wouldn't block it but want to point out that there seems to be quite a small span between 1.14 (8/2018) and 1.19 (1/2019) that having multiple instrumentation seems a bit overkill.

those dates don't look right to me: https://search.maven.org/artifact/com.azure/azure-core

**Comment by anuraaga** [comment]
Hmm, I got it from mvnrepository.com which surprisingly is more accurate it seems

https://mvnrepository.com/artifact/com.microsoft.azure/azure/1.14.0
https://repo1.maven.org/maven2/com/microsoft/azure/azure/

**Comment by trask** [comment]
oh, you're looking at the groupId/artifactId

**Comment by anuraaga** [comment]
Oops that explains it. So I guess 1.14 being less than a year old, so pretty new, is motivation to instrument it that seems reasonable.

**Comment by trask** [comment]
Ooops, I forgot was going to convert tests to Java, will send separate PR.

**Pull request merged**